src/renderer/component/wunderbar/view.jsx

```diff
@@ -1,166 +1,116 @@
+    (this: any).handleChange = this.handleChange.bind(this);
+    (this: any).focus = this.focus.bind(this);
+    this.input = undefined;
   }
 
-  componentWillUnmount() {
-    if (this.userTypingTimer) {
-      clearTimeout(this._userTypingTimer);
-    }
-  }
-
-  onChange(event) {
-    if (this._userTypingTimer) {
-      clearTimeout(this._userTypingTimer);
-    }
+  input: ?HTMLInputElement;
 
-    this.setState({ address: event.target.value });
-
-    this._isSearchDispatchPending = true;
-
-    const searchQuery = event.target.value;
-
-    this._userTypingTimer = setTimeout(() => {
-      const hasQuery = searchQuery.length === 0;
-      this._resetOnNextBlur = hasQuery;
-      this._isSearchDispatchPending = false;
-      if (searchQuery) {
-        this.props.onSearch(searchQuery.trim());
-      }
-    }, WunderBar.TYPING_TIMEOUT); // 800ms delay, tweak for faster/slower
-  }
+  handleChange(e: SyntheticInputEvent<*>) {
+    const { updateSearchQuery, getSearchSuggestions } = this.props;
+    const { value } = e.target;
 
-  componentWillReceiveProps(nextProps) {
-    if (
-      nextProps.viewingPage !== this.props.viewingPage ||
-      nextProps.address != this.props.address
-    ) {
-      this.setState({ address: nextProps.address, icon: nextProps.icon });
-    }
+    updateSearchQuery(value);
+    getSearchSuggestions(value);
   }
 
-  onFocus() {
-    this._stateBeforeSearch = this.state;
-    const newState = {
-      icon: 'icon-search',
-      isActive: true,
-    };
-
-    this._focusPending = true;
-    // below is hacking, improved when we have proper routing
-    if (!this.state.address.startsWith('lbry://') && this.state.icon !== 'icon-search') {
-      // onFocus, if they are not on an exact URL or a search page, clear the bar
-      newState.address = '';
+  focus() {
+    const { input } = this;
+    if (input) {
+      input.focus();
     }
-    this.setState(newState);
   }
 
-  onBlur() {
-    if (this._isSearchDispatchPending) {
-      setTimeout(() => {
-        this.onBlur();
-      }, WunderBar.TYPING_TIMEOUT + 1);
-    } else {
-      const commonState = { isActive: false };
-      if (this._resetOnNextBlur) {
-        this.setState(Object.assign({}, this._stateBeforeSearch, commonState));
-        this._input.value = this.state.address;
-      } else {
-        this._resetOnNextBlur = true;
-        this._stateBeforeSearch = this.state;
-        this.setState(commonState);
-      }
+  handleSubmit(value: string) {
+    if (!value) {
+      return;
     }
-  }
 
-  componentDidUpdate() {
-    if (this._input) {
-      const start = this._input.selectionStart,
-        end = this._input.selectionEnd;
+    const { onSubmit, onSearch } = this.props;
 
-      this._input.value = this.state.address; // this causes cursor to go to end of input
+    // if they choose the "search for {value}" in the suggestions
+    // it will contain the {query}?search
+    const choseDoSuggestedSearch = value.endsWith('?search');
```

Comment:
I'm not sure I like this, we need to choose to `onSearch` based on some flag for the `Search for "{query}"` suggestion